hudi-sync/hudi-hive-sync/src/main/java/org/apache/hudi/hive/HiveSyncTool.java

```diff
@@ -325,13 +318,19 @@ private boolean syncSchema(String tableName, boolean tableExists, boolean useRea
         // Sync the table properties if the schema has changed
         if (config.getString(HIVE_TABLE_PROPERTIES) != null || config.getBoolean(HIVE_SYNC_AS_DATA_SOURCE_TABLE)) {
           syncClient.updateTableProperties(tableName, tableProperties);
-          syncClient.updateSerdeProperties(tableName, serdeProperties);
+          syncClient.updateStorageDescriptor(tableName, serdeProperties, null);
           LOG.info("Sync table properties for " + tableName + ", table properties is: " + tableProperties);
         }
         schemaChanged = true;
       } else {
         LOG.info("No Schema difference for " + tableName);
       }
+
+      if (hiveSyncTableStrategy != null && !hiveSyncTableStrategy.equalsIgnoreCase(HoodieSyncTableStrategy.ALL.name())) {
```

Comment:
as @lokeshj1703 mentioned, the update should be applicable to any table regardless of sync strategy